src/components/sets/TreeNode.js

```diff
@@ -0,0 +1,203 @@
+  renderSelector = () => {
+    const {
+      title,
+      size,
+      isCurrentSet,
+      isSelected,
+      isEditing,
+    } = this.props;
+    const {
+      rcTree: {
+        prefixCls,
+        draggable,
+      },
+    } = this.context;
+
+    const wrapClass = `${prefixCls}-node-content-wrapper`;
+
+    return (
+      <span
+        ref={this.setSelectHandle}
+        title={typeof title === 'string' ? title : ''}
+        className={classNames(
+          `${wrapClass}`,
+          `${wrapClass}-${this.getNodeState() || 'normal'}`,
+          isSelected && `${prefixCls}-node-selected`,
+          (!isCurrentSet && !isEditing && draggable) && 'draggable',
+        )}
+        draggable={(!isCurrentSet && !isEditing && draggable) || undefined}
+        aria-grabbed={(!isCurrentSet && !isEditing && draggable) || undefined}
+        onDragStart={(!isCurrentSet && !isEditing && draggable) ? this.onDragStart : undefined}
+      >
+        {isCurrentSet ? (
+          <CurrentSetNode {...this.props} prefixCls={prefixCls} />
+        ) : (
+          <NamedSetNode {...this.props} prefixCls={prefixCls} />
+        )}
+        <span className={`${prefixCls}-set-size`}>{size}</span>
+      </span>
+    );
+  };
+
+  render() {
+    const { loading } = this.props;
+    const {
+      className, style,
+      dragOver, dragOverGapTop, dragOverGapBottom,
+      isLeaf,
+      expanded, selected, checked, halfChecked,
+      ...otherProps
+    } = this.props;
+    const {
+      rcTree: {
+        prefixCls,
+        filterTreeNode,
+        draggable,
+      },
+    } = this.context;
+    const disabled = this.isDisabled();
+    const dataOrAriaAttributeProps = getDataAndAria(otherProps);
+    return (
+      <li
+        className={classNames(className, {
+          [`${prefixCls}-treenode-disabled`]: disabled,
+          [`${prefixCls}-treenode-switcher-${expanded ? 'open' : 'close'}`]: !isLeaf,
+          [`${prefixCls}-treenode-checkbox-checked`]: checked,
+          [`${prefixCls}-treenode-checkbox-indeterminate`]: halfChecked,
+          [`${prefixCls}-treenode-selected`]: selected,
+          [`${prefixCls}-treenode-loading`]: loading,
+
+          'drag-over': !disabled && dragOver,
+          'drag-over-gap-top': !disabled && dragOverGapTop,
+          'drag-over-gap-bottom': !disabled && dragOverGapBottom,
+          'filter-node': filterTreeNode && filterTreeNode(this),
+        })}
+        style={style}
+        role="treeitem"
+        onDragEnter={draggable ? this.onDragEnter : undefined}
+        onDragOver={draggable ? this.onDragOver : undefined}
+        onDragLeave={draggable ? this.onDragLeave : undefined}
+        onDrop={draggable ? this.onDrop : undefined}
+        onDragEnd={draggable ? this.onDragEnd : undefined}
+        {...dataOrAriaAttributeProps}
+      >
+        {this.renderSwitcher()}
+        {this.renderCheckbox()}
+        {this.renderSelector()}
+        {this.renderChildren()}
```

Comment:
I agree that it may be more readable but those functions are all inherited meaning I would have to do something like this which I do not think makes much sense
```
  Switcher() {
      return this.renderSelector();
  }

  Checkbox() {
      return this.renderCheckbox();
  }

  Selector() {
      return this.renderSelector();
  }

  Children() {
      return this.renderChildren();
  }

  
```